HB/structure.elpi

```diff
@@ -508,6 +516,19 @@ main-declare-structure Module GRFSwP ClosureCheck :- std.do! [
   if-verbose (coq.say "HB: stop module Exports"),
   log.coq.env.end-module-name "Exports" Exports,
 
+  log.coq.env.import-module Exports,
+
+  if-verbose (coq.say "HB: declaring class abbreviation"),
+
+  mk-infer-key ClassProjection NilwP (global Structure) PhClass,
+  mk-phant-abbrev "of_" PhClass _ ClassAbbrev,
+  @global! => log.coq.notation.add-abbreviation "Build" 2
+    {{fun T C => ((lp:{{ { coq.notation.abbreviation ClassAbbrev [{{C}}] } }})
+        : (lp:{{ {mk-app (global ClassName) {params->holes NilwP} } }} T)
+      ) }} tt ClassForAbbrev,
+  @global! => log.coq.notation.add-abbreviation "of" 1
+    {{fun T => lp:{{ { coq.notation.abbreviation ClassForAbbrev [{{T}}, {{T}}] } }}}} tt _,
```

Comment:
it takes a PhD to understand this, I think you can
```
(pi x\ coq.notation.abbreviation ClassForAbbrev [x,x] (A x)),
@global! => log.coq.notation.add-abbreviation "of" 1 {{ fun t => lp:(A t) }},
```

Reply:
Sure, I can do that. Aren't you happy that your implementation of splitting under quotation/anti-quotation works so well though?

Reply:
Let's put it like that, one less thing to explain next week.